### ISSUE TITLE
Enable Link-Time Optimization (LTO)

### ISSUE THREAD

**Issue by zamazan4ik** [fujiapple852/trippy#1341]
**Describe the feature you'd like**
I noticed that in the `Cargo.toml` file Link-Time Optimization (LTO) for the project is not enabled. I suggest switching it on since it will reduce the binary size (always a good thing to have) and will likely improve the application's performance a bit.

I suggest enabling LTO only for the Release builds so as not to sacrifice the developers' experience while working on the project since LTO consumes an additional amount of time to finish the compilation routine. If you think that a regular Release build should not be affected by such a change as well, then I suggest adding an additional `dist` or `release-lto` profile where additionally to regular `release` optimizations LTO will also be added. Such a change simplifies life for maintainers and others interested in the project persons who want to build the most performant version of the application. Using ThinLTO should also help to reduce the build-time overhead with LTO. If we enable it on the Cargo profile level, users, who install the application with `cargo install`, will get the LTO-optimized version "automatically". E.g., check `cargo-outdated` Release [profile](https://github.com/kbknapp/cargo-outdated/blob/master/Cargo.toml#L48).

Basically, it can be enabled with the following lines:

```
[profile.release]
lto = true
```

I already found a related [comment](https://github.com/fujiapple852/trippy/pull/861#issuecomment-1856086104) - seems like you are not against enabling LTO.

**Describe alternatives you've considered**
Leave things as is.

**Additional context**
N/A

**Comment by c-git** [comment]
Hi, thank you for the idea. I did a quick search but didn't turn up a conclusive answer (will still do a more through one, but just checking based on your experience). Do you know if the only downside is longer compile times for release builds?

**Comment by zamazan4ik** [comment]
> Do you know if the only downside is longer compile times for release builds?

Sometimes you can meet specific build issues (nothing in common in my experience - they are unique in each situation). Sometimes there are "interesting" compiler bugs but they are quite rare nowadays in the LTO field.

So I would say no - you shouldn't expect issues with LTO except increased build time. And even this issue can be mitigated by using Thin LTO instead of Fat LTO - Thin LTO is a bit less efficient than Fat from the performed compiler optimization perspective but has far less build time overhead (btw Google by default uses ThinLTO in their prod env).

**Comment by fujiapple852** [comment]
Thanks for the suggestion @zamazan4ik. 

I don't expect much real world benefit from this as Trippy is not CPU bound, it spends most of its time waiting for packets to arrive, but I'd welcome a PR which sets LTO to `true` or `thin` (for the release profile only) none the less.  

I don't tend up build `release` builds locally except as a test before releases, but I would like to see a benchmark of build times (incremental and clean builds) with this change.

Aside: I'm also tempted to enable [`debug`](https://doc.rust-lang.org/cargo/reference/profiles.html#debug) symbols for release builds, but it hasn't proven necessary so far so perhaps a decision for another day.

**Comment by zamazan4ik** [comment]
Test env:
* Fedora 40
* Rustc 1.81
* Latest `trippy` sources
* AMD Ryzen 9 5900x

Build times (clean build):
* Regular release: 20s
* Fat LTO: 53s

Binary sizes:
* Regular release: 12 Mib
* Fat LTO: 8.6 Mib

Didn't test ThinLTO though.

**Comment by zamazan4ik** [comment]
Sure: https://github.com/fujiapple852/trippy/pull/1352
